pkg/virtctl/credentials/credentials.go

```diff
@@ -0,0 +1,456 @@
+			}
+
+			accessCredentials, err := getAccessCredentialsForVm(cmd.Context(), cli, vmNamespace, vmName)
+			if err != nil {
+				return fmt.Errorf("error getting access credentials: %w", err)
+			}
+
+			secrets := getSshSecretsForUser(accessCredentials, cmdFlags.user)
+			if len(secrets) == 0 {
+				return fmt.Errorf("no secrets specified for user: %s", cmdFlags.user)
+			}
+
+			secretName := cmdFlags.secret
+			if secretName == "" {
+				if len(secrets) > 1 {
+					return fmt.Errorf("multiple secrets specified for user %s, use option --secret to specify which secret to modify", cmdFlags.secret)
+				}
+				secretName = secrets[0]
+			} else {
+				var found bool
+				for _, secret := range secrets {
+					if secretName == secret {
+						found = true
+						break
+					}
+				}
+				if !found {
+					return fmt.Errorf("user %s does not have assigned secret %s", cmdFlags.user, cmdFlags.secret)
+				}
+			}
+
+			secret, err := cli.CoreV1().Secrets(vmNamespace).Get(cmd.Context(), secretName, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("error getting secret \"%s\": %w", secretName, err)
+			}
+
+			if secretContainsKey(secret.Data, sshKey) {
+				cmd.Printf("Secret \"%s\" already contains this SSH key.", secretName)
+				return nil
+			}
+
+			randomFilename := "ssh-key-" + rand.String(6)
+			patchData := patchToAddDataToSecret(randomFilename, []byte(sshKey))
+
+			_, err = cli.CoreV1().Secrets(vmNamespace).Patch(cmd.Context(), secretName, types.JSONPatchType, patchData, metav1.PatchOptions{})
+			if err != nil {
+				return fmt.Errorf("error patching secret \"%s\": %w", secretName, err)
+			}
+
+			cmd.Printf("Successfully added the key to secret \"%s\"", secretName)
+			return nil
+		},
+	}
+	cmdFlags.AddToCommand(cmd)
+
+	return cmd
+}
+
+func newRemoveKeyCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmdFlags := &sshCommandFlags{}
+	cmd := &cobra.Command{
+		Use:   "remove-ssh-key",
+		Short: "Remove credentials from a virtual machine.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			vmName := args[0]
+
+			vmNamespace, _, err := clientConfig.Namespace()
+			if err != nil {
+				return fmt.Errorf("error getting namespace: %w", err)
+			}
+
+			// Reading the key before accessing cluster
+			sshKey, err := getSshKey(cmdFlags)
+			if err != nil {
+				return fmt.Errorf("error getting ssh key: %w", err)
+			}
+
+			cli, err := kubecli.GetKubevirtClientFromClientConfig(clientConfig)
+			if err != nil {
+				return fmt.Errorf("error getting kubevirt client: %w", err)
+			}
+
+			accessCredentials, err := getAccessCredentialsForVm(cmd.Context(), cli, vmNamespace, vmName)
+			if err != nil {
+				return fmt.Errorf("error getting access credentials: %w", err)
+			}
+
+			secrets := getSshSecretsForUser(accessCredentials, cmdFlags.user)
+			if len(secrets) == 0 {
+				cmd.Printf("No secrets associated with user %s", cmdFlags.user)
+				return nil
+			}
```

Comment:
nit: from the beginning up to this point you are repeating the same code for adding and removing the key, can we put it into a function?

Reply:
done